internal/layer2/mime.go

```diff
@@ -0,0 +1,53 @@
+package layer2
+
+import (
+	"bytes"
+	"fmt"
+	"mime/multipart"
+	"net/textproto"
+	"strings"
+)
+
+var (
+	cloudConfigType = textproto.MIMEHeader{
+		"content-type": {"text/cloud-config"},
+	}
+
+	multipartHeader = strings.Join([]string{
+		"MIME-Version: 1.0",
+		"Content-Type: multipart/mixed; boundary=\"%s\"",
+		"\n",
+	}, "\n")
+)
+
+// GenerateInitDocument generates a multipart document with cloud-config data
+func GenerateInitDocument(rawUserData, layer2UserData string) (string, error) {
```

Comment:
let's make this more generic, something like append input user data and multiple clould config could be added as deemed necessary